### ISSUE TITLE
Remove BackingCollection from Storage

### ISSUE THREAD

**Issue by barncastle** [wowdev/DBCD#7]
Was only added as a hackfix for DBC diffing on WoW.Tools. This really needs to be fixed in the diff lib or WoW.Tools and removed from here.

[RE: Here](https://github.com/wowdev/DBCD/blob/master/DBCD/DBCDStorage.cs#L65)

**Comment by Marlamin** [comment]
Any suggestions on how we should go about doing this?

**Comment by barncastle** [comment]
Should be fixed in 0f7be03ab0f490a70da61705008204ddfe5d88d8, turns out overriding `GetDynamicMemberNames` was the last piece of the `DynamicObject` puzzle.

As for DBCDumpHost , we need to change the `DiffController` to the following:
```c#
var dbc1 = (IDictionary)dbcManager.GetOrLoad(name, build1, useHotfixesFor1);
var dbc2 = (IDictionary)dbcManager.GetOrLoad(name, build2, useHotfixesFor2);
```
I'll create a PR tonight if you want?
Also is it possible to test the performance difference of this on the site as I think it'll be slower?

**Comment by Marlamin** [comment]
Should be possible yeah, can test performance locally as well. As for PR that's fine, but DBCDumpHost is currently still using the hotfix branch, will need updating to master too.
